src/core/lv_refr.c

```diff
@@ -611,18 +611,19 @@ static void refr_area(const lv_area_t * area_p)
         layer->buf_area.y1 = 0;
         layer->buf_area.x2 = lv_display_get_horizontal_resolution(disp_refr) - 1;
         layer->buf_area.y2 = lv_display_get_vertical_resolution(disp_refr) - 1;
-        layer_reshape_draw_buf(layer);
         lv_area_t disp_area;
         lv_area_set(&disp_area, 0, 0, lv_display_get_horizontal_resolution(disp_refr) - 1,
                     lv_display_get_vertical_resolution(disp_refr) - 1);
 
         if(disp_refr->render_mode == LV_DISPLAY_RENDER_MODE_FULL) {
             disp_refr->last_part = 1;
+            layer_reshape_draw_buf(layer, layer->draw_buf->header.stride);
```

Comment:
Because need to distinguish between using `LV_STRIDE_AUTO` and the original value.
When using DIRECT/FULL rendering mode, you need to maintain the stride set by the user.
When using PARTIAL, the render buffer will automatically adjust the size, which needs to be automatically calculated using `LV_STRIDE_AUTO`.

Reply:
Makes sense.